tools/integration_tests/util/client/storage_client.go

```diff
@@ -19,25 +19,56 @@ import (
 	"fmt"
 	"io"
 	"log"
+	"os"
 	"reflect"
 	"testing"
 	"time"
 
 	"cloud.google.com/go/storage"
 	"github.com/googlecloudplatform/gcsfuse/v2/tools/integration_tests/util/operations"
 	"github.com/googlecloudplatform/gcsfuse/v2/tools/integration_tests/util/setup"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
 	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+	storagev1 "google.golang.org/api/storage/v1"
 )
 
-func CreateStorageClient(ctx context.Context) (*storage.Client, error) {
+func CreateStorageClient(ctx context.Context) (client *storage.Client, err error) {
 	// Create new storage client.
-	client, err := storage.NewClient(ctx)
+	if setup.TestOnTPCEndPoint() {
+		var ts oauth2.TokenSource
+		// Set up the TPC endpoint and provide a token source for authentication.
+		ts, err = getTokenSrc("/tmp/sa.key.json")
+		if err != nil {
+			return nil, fmt.Errorf(fmt.Sprintf("Unable to fetch tokenSrc for TPC: %v", err))
```

Comment:
nit: can we replace fmt.Errorf with
fmt.Errorf("unable to fetch token-source for TPC: %w", err)

because:
* it's concise: no need to call fmt.Sprintf
* errors are supposed to start with lowercase letters
* token-source is more readable than token-source
* Replace %v with %w [reasoning here](https://go.dev/wiki/ErrorValueFAQ#i-am-already-using-fmterrorf-with-v-or-s-to-provide-context-for-an-error-when-should-i-switch-to-w)